### PR TITLE
Handle empty array in ExtractViewHelper

### LINKED ISSUE #1451: ExtractViewHelper: returns NULL for empty array
If the `ExtractViewHelper` is given an empty array with an arbitrary `key`, it always returns `NULL` instead of an empty array.

### PR DESCRIPTION
**Pull request by mbrodala**
Fixes #1451

### CODE REVIEW

**Comment by NamelessCoder** [comment]
This looks appropriate, but changes a return type from null to array. That's okay but there appears to be at least one (more) test that needs changing.

**Comment by mbrodala** [comment]
You are right, I didn't take care of the non-recursive variant.

**Comment by mbrodala** [comment]
The tests are fixed now.

File: Classes/ViewHelpers/Iterator/ExtractViewHelper.php [hunk 1]
```diff
@@ -172,7 +172,7 @@ protected static function recursivelyExtractKey($iterator, $key)
         if (false === is_array($iterator) && false === $iterator instanceof \Traversable) {
             throw new \Exception('Traversable object or array expected but received ' . gettype($iterator), 1515498714);
         }
-
```
Comment (NamelessCoder) [inline Classes/ViewHelpers/Iterator/ExtractViewHelper.php]: Eew, how did that get in there? :)

**Pull request merged**